torch/testing/_internal/common_methods_invocations.py

```diff
@@ -5423,10 +5423,23 @@ def sample_inputs_fmod_remainder(op_info, device, dtype, requires_grad, *, autod
     make_arg = partial(make_tensor, dtype=dtype, device=device, requires_grad=requires_grad)
 
     if autodiffed:
-        samples = (
-            ((S, S, S), 1.5, False),
-            ((), 1.5, False),
-        )
+        if dtype.is_floating_point or dtype.is_complex:
+            samples = (
+                ((S, S, S), 1.5, False),
+                ((), 1.5, False),
+            )
+        else:
+            # NOTE: Given an integer tensor and a non-tensor number, fmod and
```

Comment:
Just wanted to draw attention to this oddity. Maybe there's a better way to handle it than this?

Reply:
Do we still want them to return a double tensor? Like what if we just let them return a float32 tensor in these cases?